### ISSUE TITLE
Upsert query variables fails

### ISSUE THREAD

**Issue by Q-Minh** [dgraph-io/dgraph-js#93]
### Working Upsert

```
const query = `
    query {
        user as q(func: eq(user.id, "${request.userId}"))
    }
    `;

const mu = new dgraph.Mutation();
mu.setDelNquads(`uid(user) * * .`);

const req = new dgraph.Request();
req.setQuery(query);
req.setMutationsList([mu]);
req.setCommitNow(true);

const res = await dgraphClient.newTxn().doRequest(req);
```

When directly inserting the value of `request.userId` into the query string via string interpolation, the node returned by the query is successfully deleted.

### Failing upsert

```
const query = `
            query q($userId: string) {
                user as q(func: eq(user.id, $userId))
            }
        `;
    
const mu = new dgraph.Mutation();
mu.setDelNquads(`uid(user) * * .`);
    
const req = new dgraph.Request();
const vars = req.getVarsMap();
vars.set("$userId", `${request.userId}`);
req.setQuery(query);
req.setMutationsList([mu]);
req.setCommitNow(true);
    
const res = await dgraphClient.newTxn().doRequest(req);
```

When trying to use query variables and setting the corresponding entry in the variables map for `$userId` with `` `${request.userId}` ``, the node that should be returned by the query is not successfully deleted. 

Is this an unsupported feature for upserts or is something wrong with the code?

**Comment by prashant-shahi** [comment]
Are you using the **Type System** for that node in the schema?

If the *type information* is missing, the delete operation will be a *no-op*.

**Comment by Q-Minh** [comment]
Yes, I am using the Type System for that node. If I returned its predicate dgraph.type, I can see its type. The "set" mutations work in upserts with query variables, only the deletions seem to be no-ops.

**Comment by prashant-shahi** [comment]
I was able to reproduce the issue that you have raised.
I also saw similar behavior in Python client, so I believe it is a bug related to Dgraph.

Thanks, @Q-Minh, for creating this issue. Transfering it to Dgraph.
